### PR TITLE
Add additional info in Mock's Change

### PR DESCRIPTION
**Pull request by Toa741**

### CODE REVIEW

File: NGitLab.Mock/MergeRequest.cs [hunk 1]
```diff
@@ -80,13 +80,12 @@ public MergeRequestChangeCollection Changes
         {
             get
             {
-                var tipCommit = SourceProject.Repository.GetBranchTipCommit(SourceBranch);
                 var stats = Project.Repository.GetBranchFullPatch(SourceBranch);
                 var changes = new MergeRequestChangeCollection(this);
                 foreach (var stat in stats)
                 {
                     var diff = stat.Patch.Substring(stat.Patch.IndexOf("@@", StringComparison.InvariantCulture));
```
Comment (meziantou) [inline NGitLab.Mock/MergeRequest.cs]:
```suggestion
                    var diff = stat.Patch.Substring(stat.Patch.IndexOf("@@", StringComparison.Ordinal));
```

File: NGitLab.Mock/MergeRequestChangeCollection.cs [hunk 2]
```diff
@@ -10,11 +11,25 @@ public MergeRequestChangeCollection(GitLabObject parent)
         }
 
         public Change Add(string diff)
```
Comment (meziantou) [inline NGitLab.Mock/MergeRequestChangeCollection.cs:14]: Should we obsolete this method?

File: NGitLab.Mock/MergeRequestChangeCollection.cs [hunk 3]
```diff
@@ -10,11 +11,25 @@ public MergeRequestChangeCollection(GitLabObject parent)
...
             if (diff is null)
                 throw new ArgumentNullException(nameof(diff));
 
-            var change = new Change() { Diff = diff };
+            var change = new Change
+            {
+                Diff = diff,
+                NewPath = newPath,
+                OldPath = oldPath,
+                DeletedFile = changeKind.HasValue && changeKind.Value == ChangeKind.Deleted,
```
Comment (louis-z) [inline NGitLab.Mock/MergeRequestChangeCollection.cs]: You can simplify the condition as follows (same for other 2)
```suggestion
                DeletedFile = changeKind.Value == ChangeKind.Deleted,
```

Comment (Toa741) [reply NGitLab.Mock/MergeRequestChangeCollection.cs]: changeKind is nullable, wouldn't it trigger an exception?

File: NGitLab.Mock/MergeRequestChangeCollection.cs (hunk 2 above)
Comment (Toa741) [reply NGitLab.Mock/MergeRequestChangeCollection.cs:14]: Yes, would make things better in the end

File: NGitLab.Mock/MergeRequestChangeCollection.cs (hunk 3 above)
Comment (meziantou) [reply NGitLab.Mock/MergeRequestChangeCollection.cs]: Nope, `T? item1 == T item2` is equivalent to `item1.HasValue && item1.Value == item2`

Comment (Toa741) [reply NGitLab.Mock/MergeRequestChangeCollection.cs]: Oh, I think it's just the suggestion given by louis:
`changeKind.Value == ChangeKind.Deleted` => `changeKind== ChangeKind.Deleted`

Will update that.

File: NGitLab.Mock/MergeRequest.cs (hunk 1 above)
Comment (Toa741) [reply NGitLab.Mock/MergeRequest.cs]: Updated.

**Pull request merged**